src/rp2_common/hardware_pio/pio.c

```diff
@@ -50,6 +51,12 @@ int pio_claim_unused_sm(PIO pio, bool required) {
     return index >= (int)base ? index - (int)base : -1;
 }
 
+bool pio_sm_is_claimed(PIO pio, uint sm) {
+    check_sm_param(sm);
+    uint which = pio_get_index(pio);
+    return hw_is_claimed(&claimed, which * NUM_PIO_STATE_MACHINES + sm);
```

Comment:
Might be worth adding a `static inline uint pio_sm_bit(PIO pio, uint sm)` similar to the `static inline uint interp_lane_bit(interp_hw_t * interp, uint lane)` ?

Reply:
yeah, there are about 4 different calculations that need to b e in sync, only two share, so not too worried